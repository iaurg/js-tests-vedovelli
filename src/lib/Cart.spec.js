import Cart from './Cart';

describe('Cart', () => {
  let cart;

  beforeEach(() => {
    cart = new Cart();
  });

  it('should return 0 when call getTotal() in a newly created cart', () => {
    expect(cart.getTotal()).toEqual(0);
  });

  it('should multiply quantity and price and receive the total amount of cart', () => {
    const item = {
      product: {
        title: 'Cooler Titanium Max',
        price: 35190,
      },
      quantity: 2,
    };

    cart.addItem(item);

    expect(cart.getTotal()).toEqual(70380);
  });
});
