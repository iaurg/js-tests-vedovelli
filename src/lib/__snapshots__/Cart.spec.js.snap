// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() should return a object with the total and the list of items in cart 1`] = `
Object {
  "items": Array [
    Object {
      "product": Object {
        "price": 35190,
        "title": "Cooler Titanium Max",
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "price": 10090,
        "title": "Air Max",
      },
      "quantity": 1,
    },
  ],
  "total": 80470,
}
`;
